### ISSUE TITLE
Bug DAX Editor

### ISSUE THREAD

**Issue by PalombieChamonix** [TabularEditor/TabularEditor3#857]
The DAX editor has a strange behevior since the last release, which makes it very frustrating to use !! 
![2023-03-21_18-14-27](https://user-images.githubusercontent.com/91263359/226689942-d34a14d4-dab0-4db5-8325-b16e20778625.gif)

**Comment by jbrizido** [comment]
Happens to me too. Although it does not happen in all reports.

**Comment by mlonsk** [comment]
Hi @PalombieChamonix 
Are you working in the expression editor or in a DAX Script and does it happen in both?

**Comment by mlonsk** [comment]
Hi @PalombieChamonix , @jbrizido & @BitwalkerSthlm 

I am having a bit of trouble reproducing the issue in the GIF. Would it be possible to share some DAX code as well as a description of each step taken to trigger the bug?

**Comment by BitwalkerSthlm** [comment]
I only use the Expression Editor, so that's where the issue occurred for me. I reverted back to the previous version, so I can't record anything, but for instance it happened when I positioned the cursor at the start of the row above RETURN in the measure below and hit enter. 

```
VAR VAR_MinDate = MIN('DIM Dates'[Day Date])
VAR VAR_MaxDate = MAX('DIM Dates'[Day Date])
VAR VAR_ReEmployLimit = [Re-employment Limit]

VAR VAR_Employees =  
      
    CALCULATETABLE(
        SUMMARIZE(
            FILTER( 'DIM Employment Details',
                
                --Has ended row
                'DIM Employment Details'[Row Valid To] >= VAR_MinDate &&
                'DIM Employment Details'[Row Valid To] <= VAR_MaxDate &&
                
				--Räkna bara de som avslutat en fast anställning
				'DIM Employment Details'[Employment Basic Type] = "Permanent Employment" &&
				
				--Ingen senare fast anställning samma år
				'DIM Employment Details'[E Following Permanent Rows Same Year] = FALSE &&
				
				--Now other employments which interfere
				(
					'DIM Employment Details'[E Has Future Fixed-term Row] = FALSE ||
					'DIM Employment Details'[E Has Future Fixed-term Row] = TRUE && 'DIM Employment Details'[E Gap to Future Fixed-term Row] >= VAR_ReEmployLimit
				) &&
				(
					'DIM Employment Details'[E Has Future Permanent Row] = FALSE ||
					'DIM Employment Details'[E Has Future Permanent Row] = TRUE && 'DIM Employment Details'[E Gap to Future Permanent Row] >= VAR_ReEmployLimit
				)
                
            )
            , 'DIM Employment Details'[Employee Id]
        )
        , TREATAS( VALUES( 'DIM Organization'[Cost Center MDS Code] ), 'DIM Employment Details'[Cost Center MDS Code] )
        , TREATAS( VALUES( 'DIM Employees'[Employee Id] ) ,'DIM Employment Details'[Employee Id])
    )

RETURN
    
    IF(
        ISFILTERED( 'DIM Dates' ) = FALSE,
        
        BLANK(),
        
        CALCULATE( 
            DISTINCTCOUNT('DIM Employees'[Employee Id]), 
            TREATAS( VAR_Employees, 'DIM Employees'[Employee Id] )
        
    )
    )
```

**Comment by otykier** [comment]
Great repro, thanks @BitwalkerSthlm!

**Comment by otykier** [comment]
**Update:** We're working on fixing this issue. In the mean time, you can prevent the issue by diabling "Auto-indent" in the **Tools > Preferences > Text Editors > DAX Editor > Auto Formatting** page.

![image](https://user-images.githubusercontent.com/8976200/227478466-35d1a6be-fda1-4811-8a54-0df6bd68e8fa.png)

**Comment by BitwalkerSthlm** [comment]
Great, thanks!!

**Comment by otykier** [comment]
This is fixed in [3.5.1](https://docs.tabulareditor.com/te3/other/release-notes/3_5_1.html?tabs=downloads). After installing 3.5.1, the issue should no longer appear, even when re-enabling **Auto-indent**.

**Comment by CraigsterT** [comment]
There seems to be a new issue in 3.5.1. When I am typing out a simple
measure , for instance , Sales YTD , I start the measure CALCULATE( [Sales
Amount] , and at this stage the closing parenthesis for CALCULATE()
suddenly vanishes and has to be retyped. Is this a bug in the editor ?
